util/src/main/scala/com/scalableminds/util/requestlogging/RequestLogging.scala

```diff
@@ -0,0 +1,38 @@
+package com.scalableminds.util.requestlogging
+
+import com.typesafe.scalalogging.LazyLogging
+import play.api.http.HttpEntity
+import play.api.mvc.{Request, Result}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+trait AbstractRequestLogging extends LazyLogging {
+
+  def logRequestFormatted(request: Request[_], result: Result, userMail: Option[String] = None) = {
+    val emailMsg = userMail.map(m => s" for $m").getOrElse("")
+    result.body match {
+      case HttpEntity.Strict(byteString, contentType) if result.header.status != 200 =>
+        logger.warn(s"Answering ${result.header.status} at ${request.uri}$emailMsg – ${byteString.take(20000).decodeString("utf-8")}")
```

Comment:
Is `logger.warn` enough or would `logger.error` be better?

Reply:
Thought about this again and decided that `warn` is better for now, because some of this might be expected. It does not automatically mean that there is something wrong with wK if one of these “errors” happens. I’m happy to discuss further if you disagree.